aerosonicdb/features/extract_mfccs.py

```diff
@@ -28,115 +33,138 @@
-
-        data = {
-            'mfcc': [],
-            'class_label': [],
-            'subclass_label': [],
-            'fold_label': []
-        }
-
-        df = pd.read_csv(meta_path)
-
-        df = df[df['train-test'] == set_str].reset_index(drop=True)
-
-        for row in df.index:
-            base_filename = df['filename'].iloc[row]
-
-            audio_path = get_audio_path(df=df, filename=base_filename, dataset_path=dataset_path)
-
-            class_label = df['class'].iloc[row]
-            subclass_label = df['subclass'].iloc[row]
-            fold_label = df['fold'].iloc[row]
-            offset = df['offset'].iloc[row]
-            clip_duration = df['duration'].iloc[row]
-
-            signal, sr = librosa.load(audio_path, offset=offset, duration=clip_duration)
-
-            if sr == sample_rate:
-
-                # data['filename'].append(base_filename)
-
-                clip_segments = int(np.ceil(len(signal) / SAMPLES_PER_SEGMENT))
-
-                for s in range(clip_segments):
-                    start = SAMPLES_PER_SEGMENT * s
-                    end = start + SAMPLES_PER_SEGMENT
-
-                    if len(signal[start:]) < SAMPLES_PER_SEGMENT:
-                        stub = signal[start:]
-                        padded = fix_length(stub, size=int(5*sr))
-                        mfcc = librosa.feature.mfcc(y=padded,
-                                                    sr=sample_rate,
-                                                    n_mfcc=n_mfcc,
-                                                    n_fft=n_fft,
-                                                    hop_length=hop_length)
-
-                    else:
-                        mfcc = librosa.feature.mfcc(y=signal[start:end],
-                                                    sr=sample_rate,
-                                                    n_mfcc=n_mfcc,
-                                                    n_fft=n_fft,
-                                                    hop_length=hop_length)
-
-                    mfcc = mfcc.T
-
-                    if len(mfcc) == EXPECTED_MFCC_VECTORS_PER_SEGMENT:
-                        data['mfcc'].append(mfcc.tolist())
-                        data['class_label'].append(str(class_label))
-                        data['subclass_label'].append(str(subclass_label))
-                        data['fold_label'].append(str(fold_label))
+        print(f"Output path created. Features will be saved to: {output_path}")
+
+    json_path = os.path.join(output_path, f"{n_mfcc}_mfcc_{duration}_{set_str}.json")
+
+    if os.path.exists(json_path) and not force:
+        print(f"{set_str} MFCCs extracted.")
+        return
+
+    # create a dictionary to store the JSON data
+
+    data = {"mfcc": [], "class_label": [], "subclass_label": [], "fold_label": []}
+    df = pd.read_csv(meta_path)
+    df = df[df["train-test"] == set_str].reset_index(drop=True)
+
+    for row in tqdm(df.index):
```

Comment:
wrapping in `tqdm` gives us a nice progress bar, works both for CLI and notebook invocation